project_task_category/models/project_task_category.py

```diff
@@ -11,4 +11,5 @@ class ProjectTask(models.Model):
     categ_id = fields.Many2one(
         'project.category',
         'Category',
+        domain=[('code', '=', 'categ_id')]
```

Comment:
Maybe for your particular case that's true. You can't be sure that it's always true.
If you use Projects for very different domains, sucha as IT / HR / Fleet / Facility, etc., categories usable on each will be very different.

But I can add that myself in a later PR if you prefer.